FightPandemics/Features/Post/CreatePost/TabBarUI/TabBarUI.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  TabBarUI.swift
+//  FightPandemics
+//
+//  Created by ehsan sat on 5/18/20.
+//
+//  Copyright (c) 2020 FightPandemics
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import UIKit
+
+class TabBarUI: UIView {
```

Comment:
can we call this `TabBar` instead of `TabBarUI`? no need to include `UI` in the name of a UI component

Reply:
> @ethanswift i think maybe the requirements have been lost in translation? this Issue covers replacing the tab bar of the entire app
> 
> so what i would imagine is, instead of subclassing `UIView`, you'd be subclassing `UITabBar`. and you'd have to work in `RootTabBarController` to customize its `tabBar` property
> 
> i didnt have the chance to look at your component closely - but reading it its seems as though perhaps the `dotView` is statically placed under the Home tab icon. the intention is for each tab to have a `selected` and a `deselected` state; in the `selected` state, a dot appears under the icon in place of the title
> 
> if getting the dot aspect in is a lot, i'd say its acceptable to drop the dot bit and just concern yourself with replacing the tab bar

Actually, its instance method is getting a selected enum which asks for the tab which is selected as a parameter and adds that dot and bolden that image (make it black). so in the function which is setUp(selected: .feed) you choose which tab to be loaded